src/Microsoft.ML.Core/Data/IHostEnvironment.cs

```diff
@@ -23,6 +23,28 @@ public interface IChannelProvider : IExceptionContext
         IPipe<TMessage> StartPipe<TMessage>(string name);
     }
 
+    /// <summary>
+    /// Utility class for IHostEnvironment
+    /// </summary>
+    internal static class HostEnvironmentExtensions
```

Comment:
I'm surprised this doesn't need a `[BestFriend]` attribute.  @TomFinley - do you know why?